app/services/tezos/governance_sync_service.rb

```diff
@@ -110,7 +110,9 @@ def get_proposal_and_ballot_info
 
       # If there are still unsynced blocks, run again
       if @voting_period.blocks_to_sync == []
+        puts "All blocks synced for period #{@voting_period.id}."
```

Comment:
In case of such log messages, we could use a logger that could be silenced (e.g. when running tests) instead of `puts`.

Reply:
:+1: @pienkowb, `Rails.logger.debug`.
also ` @voting_period.blocks_to_sync.empty?`, probably, on the line above